---
layout: page
title: Project
permalink: /project/
---

# Project Description

{:justify: style="text-align: justify"}
Structural rearrangements (inversions, translocations, transpositions, and large insertions, deletions and duplications) hold a key role in promoting and maintaining genetic variation with broad implications in genome instability, speciation, functional innovation and disease susceptibility. However, our understanding of their evolutionary dynamics remains limited due to insufficient resolution and accuracy in characterizing complex events with short-read sequencing. The incomplete reference genome assemblies of many organisms further exacerbate this problem, especially in repetitive and highly variable regions such as subtelomeres and chromosome-ends.
The applications of long-read sequencing technology in several recent studies have proved to be quite successful in detecting and resolving structural rearrangements, even for complex genomic regions (Chaisson et al. Nature 2014; Zapata et al. PNAS 2016; Dong et al. PNAS 2016). Here we take this to the next level, by applying long-read sequencing technology to the population level by sequencing 12 representative strains from the partially domesticated yeast *Saccharomyces cerevisiae* and its closest wild relative *Saccharomyces paradoxus*. This allowed us to systematically discover structural rearrangements based on complete genome assemblies. Furthermore, for the first time, we explicitly partitioned nuclear chromosomes into cores, subtelomeres and chromosome-ends, which allows us to assess their respective structural dynamics accordingly. In particular, instead of relying on the current subtelomere definition that treats all chromosomes indiscriminately (e.g. 20~30 kb from chromosome-ends in yeasts), we proposed a chromosome-specific subtelomere definition based on synteny conservation. Crucially, this effectively captured the dynamic nature of subtelomeres. Our high-resolution analysis of structural rearrangements across different chromosome partitions within a well-defined phylogenetic framework uncovered striking contrasts in genome dynamics between domesticated and wild yeasts. In chromosomal cores, *S. paradoxus* exhibits higher accumulation rate of balanced structural rearrangements (inversions, translocations and transpositions) whereas *S. cerevisiae* accumulates unbalanced rearrangements (large insertions, deletions and duplications) more rapidly. In subtelomeres, recurrent interchromosomal reshuffling was found in both species, with higher rate in *S. cerevisiae*. Such striking contrasts between wild and domesticated yeasts reveal the influence of human activities on structural genome evolution. In addition, we used two case studies of complex multi-allelic loci to illustrate how the precisely characterized structural rearrangements in both the core and subtelomeric regions can be linked to phenotypic diversity. Finally, for the first time, we discovered several non-canonical chromosome-end structures and lineage-specific rearrangements and introgression in mitochondrial genomes, both of which would be inaccessible without our complete genome assemblies. This work represents the third phase of our long term Saccharomyces Genome Resequencing Project (Liti et al. Nature 2009; Bergstrom et al. MBE 2014). 

{:justify}


# Impact Statement
The striking contrasts of structural genome dynamics between domesticated and wild yeasts reveal the influence of human activities on genome evolution.

# Keywords
yeast, structural rearrangement, subtelomere, evolution, genome

# Research Organisms
*Saccharomyces cerevisiae* and *Saccharomyces paradoxus*

